### PR TITLE
Adding GEOMETRY Support

### PR DESCRIPTION
**Pull request by mikeringrose**
This is to replace PR https://github.com/sequelize/sequelize/pull/3828. Same code added to the latest in master.

### CODE REVIEW

**Comment by mikeringrose** [comment]
Locally `make postgres` passes just fine.

>  1568 passing (2m)
>  2 pending

**Comment by mikeringrose** [comment]
Jan if there's anything else that'd you'd like me to change, let me know. Otherwise I'll begin adding MySQL support in another PR.

**Comment by janmeier** [comment]
@mikeringrose Nope, everything looks good! Looking forward to the next PR :)

File: lib/dialects/postgres/query-generator.js [hunk 1]
```diff
@@ -897,6 +897,13 @@ var QueryGenerator = {
       return 'ARRAY[' + value.map(function (v) {
         return SqlString.escape(JSON.stringify(v), false, this.options.timezone, this.dialect, field);
       }, this).join(',') + ']::' + jsonType.key + '[]';
+    } else if (value && field && field.type instanceof DataTypes.GEOMETRY) {
+      var s = "ST_GeomFromGeoJSON('" + JSON.stringify(value) + "'";
+      if (field.type.srid) {
+        s += + ", " + field.type.srid;
```
Comment (krisr) [inline lib/dialects/postgres/query-generator.js]: hey @mikeringrose, I think you are either missing another term in this statement, or need to remove the leading "+". I tried creating geometry with DataTypes.GEOMETRY("Point", 4326) and ended up having bad queries generated due to this line.

Comment (mikeringrose) [inline lib/dialects/postgres/query-generator.js]: Thanks for the heads up. I'll be submitting a fix for this issue tomorrow
to address the following bug -
https://github.com/sequelize/sequelize/issues/4054. If you need things to
work for now, you can leave off the SRID. Thanks again.

Mike Ringrose

**Comment by mshahriarinia** [comment]
What is the syntax for sequelize postgres geometry? Is there a HelloWorld example?

**Comment by janmeier** [comment]
@mshahriarinia We will add documentation for this once all dialects are supported.

Until then Id recommend you to look at the tests in this PR to see the syntax

**Comment by mshahriarinia** [comment]
I am using the latest version on github f38d0cae5cb3aaeb22118e0b612264aa963a2298, and I have postgres+postgis. Quick note for folks: on you DB make sure you run `CREATE EXTENSION postgis;` otherwise you get `Error: "geometry" does not exist` from postgres.

**Comment by mshahriarinia** [comment]
Can I do distance calculation between a lat/lon and a point column in potgres in current state or top k closest points to a given point? I went through all sorts of code, [looks like](https://github.com/sequelize/sequelize/pull/3699) somebody introduced the concept of `$geoWithin`, `$geoIn`, but got deleted from pull request?

**Comment by janmeier** [comment]
@mshahriarinia Unfortunately those operators were not integrated into the final pull request.

They would definitely be nice to have, so a PR to re-add them would be very wellcome! The code should be pretty similar to whats in this PR, just needs to be migrated

**Comment by mikeringrose** [comment]
That'd be awesome if we could get those in. :+1:

On Mon, Jul 13, 2015 at 3:17 AM, Jan Aagaard Meier <notifications@github.com

**Pull request merged**